plugin/sampling/strategystore/adaptive/processor_test.go

```diff
@@ -79,8 +79,15 @@ var (
 )
 
 func TestAggregateThroughput(t *testing.T) {
+	testThroughput := []*model.Throughput{
+		{Service: "svcA", Operation: "GET", Count: 4, Probabilities: map[string]struct{}{"0.1": {}}},
+		{Service: "svcA", Operation: "GET", Count: 4, Probabilities: map[string]struct{}{"0.2": {}}},
+		{Service: "svcA", Operation: "PUT", Count: 5, Probabilities: map[string]struct{}{"0.1": {}}},
+		{Service: "svcB", Operation: "GET", Count: 3, Probabilities: map[string]struct{}{"0.1": {}}},
+	}
+
 	p := &Processor{}
-	aggregatedThroughput := p.aggregateThroughput(testThroughputs)
```

Comment:
It's very likely that this global variable is the source of the flakiness as it's not `read-only` as it should therefore impacting also this case.

Reply:
I am a bit concerned by this approach. The input to `p.aggregateThroughput` is _supposed to be treated as read-only_ by that function. If it's not, that might be the bug / source of the test dependencies, not the fault of the test. The `testThroughputs` is not a global variable, it's logically a constant. By using a function instead of the same value we simply obscure the bug elsewhere.

Reply:
Let me see if I understand your point:

since method `p.aggregateThroughput` is *supposed* to treat its inputs as `read-only`, you think that my approach could hide a potential bug in the `p.aggregateThroughput` that could potentially mutate the input? 

If this is what you mean I think my change is still valid because it's enforcing independence **between** tests. And what we could have is a specific test that tests for `p.aggregateThroughput` immutability. Currently the thing that is actually mutating the content of the global variable is a test on its body (see in master `TestRealisticRunCalculationLoop`  in `processor_test:445`)

Reply:
I have created the following test case and to your point it indeed fail (which means that the method is indeed mutating the arguments):

```go
func TestAggregateThroughputInputsImmutability(t *testing.T) {
	p := &Processor{}
	in := testThroughputs()
	_ = p.aggregateThroughput(in)
	assert.Equal(t, in, testThroughputs())
}

```

Reply:
And if indeed it's supposed not to mutate the arguments the issue is exactly here:

`plugin/sampling/strategystore/adaptive/processor.go:313` 

Pointer semantics :yum: